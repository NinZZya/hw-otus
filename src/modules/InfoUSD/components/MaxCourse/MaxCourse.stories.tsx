import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { MaxCourse, MaxCourseProp } from "./MaxCourse";

export default {
  title: "Modules/InfoUSD/Components/MaxCourse",
  component: MaxCourse,
} as Meta;

export const WithValue: React.FC<MaxCourseProp> = () => <MaxCourse value={1} />;
