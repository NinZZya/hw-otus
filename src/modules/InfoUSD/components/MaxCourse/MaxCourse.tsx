import React from "react";
import { Card, Indicator, Label } from "../../../../componets";
import { getRandomColor } from "../../../../utils/utils";

export interface MaxCourseProp {
  value: number;
  indicatorColor?: string;
}

export interface MaxCourseState {
  maxValue: number;
  color: string;
}

export class MaxCourse extends React.PureComponent<
  MaxCourseProp,
  MaxCourseState
> {
  constructor(props: MaxCourseProp) {
    super(props);

    this.state = {
      maxValue: 0,
      color: "#00ff00",
    };
  }

  componentDidMount(): void {
    const { value: maxValue } = this.props;
    this.setState({ maxValue });
  }

  componentDidUpdate(prevProps: MaxCourseProp): void {
    const { value } = this.props;
    const { maxValue } = this.state;
    const isNeedUpdate = value !== prevProps.value && value > maxValue;

    if (isNeedUpdate) {
      this.setState({
        maxValue: value,
        color: getRandomColor(),
      });
    }
  }

  render(): React.ReactNode {
    const { maxValue, color } = this.state;

    return (
      <Card>
        <Indicator type={"card"} color={color} />
        <span>Max course:</span>
        <Label value={`$${maxValue}`} />
      </Card>
    );
  }
}
