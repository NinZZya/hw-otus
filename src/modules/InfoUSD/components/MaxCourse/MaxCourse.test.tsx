import React from "react";
import renderer from "react-test-renderer";
import { MaxCourse } from "./MaxCourse";

it("MaxCourse render coorectly", () => {
  const tree = renderer
    .create(<MaxCourse value={0} indicatorColor={"#000"} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
