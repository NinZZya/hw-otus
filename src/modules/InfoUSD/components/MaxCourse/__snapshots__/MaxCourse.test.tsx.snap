// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxCourse render coorectly 1`] = `
<div
  className="css-1y7yulv"
>
  <span
    className="css-1n9rj5f"
    color="#00ff00"
  >
    â€¢
  </span>
  <span>
    Max course:
  </span>
  <span
    className="css-l3kone"
  >
    $0
  </span>
</div>
`;
